internal/models/factor.go

```diff
@@ -256,3 +301,19 @@ func DeleteExpiredFactors(tx *storage.Connection, validityDuration time.Duration
 	}
 	return nil
 }
+
+func (f *Factor) FindLatestUnexpiredChallenge(tx *storage.Connection, expiryDuration float64) (*Challenge, error) {
+	now := time.Now()
+	var challenge Challenge
+
+	err := tx.Where("sent_at + interval '1 second' * ? > ?", expiryDuration, now).
+		Order("sent_at desc").
+		First(&challenge)
```

Comment:
Please add an index on `sent_at`! I'd prefer if the interval calculation is done in Go, keeping the SQL to a minimum.